mockgcp/mockapikeys/names.go

```diff
@@ -0,0 +1,56 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mockapikeys
+
+import (
+	"strings"
+
+	"github.com/GoogleCloudPlatform/k8s-config-connector/mockgcp/common/projects"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+type apiKeyName struct {
+	Project  *projects.ProjectData
+	Location string
```

Comment:
Do we always make location value configurable even if the resource is global and has only one fixed location value, `global`?

Reply:
Yes, we're parsing the value provided.  We aren't currently validating the location in any of our mocks (though we do save the object under a key which includes the location, in case we do support regions in future).